Allow bdist_wheel, support Python 3.4/3.5, make 'POOL' optional

danizen opened the pull request:
John,

I have this working now in my environment.   It works like a charm.   I have about 70 developers and more CRUD applications, and I don't want everyone of them needing to set the 'POOL' parameter.   Also, our QA and Production servers may not install from the web.   I typically build to a wheel file, which can then be distributed where needed.

If you want someone else to help maintain the package, this one doesn't seem hard to maintain.

Commit:
Changes so that this package imports properly

- Change package name to 'django_oracle_drcp' because hyphens do not
  work.
- Change setup.py so that this may be build into a wheel file.

Commit:
Cause 'POOL' to be optional since Python multiprocesses

- Oracle DRCP whitepaper indicates that min, max, increment are per-process
- Testing bears this out - max of 4 sometimes allows more connections to
  proceed
- So, for Django, typical default setting of 1, 2, 1 makes some sense.

JohnPapps commented:
Thank you for your time with this.  

Having a sensible default for the POOL makes perfect sense.  I didn't really have enough opportunity to figure out what would be sensible, so your comments are useful and welcome.

I haven't had issues with the hyphens in the package name, so would be interested to know where this has been an issue for you.

danizen commented:
> I haven't had issues with the hyphens in the package name, so would be interested to know where this has been an issue for you.

I went into python manage.py shell in one project and entered `import django-oracle-drcp` and was told there was a syntax error.   In retrospect, I can see that `__import__ ("django-oracle-drcp")` would have worked, but this seemed ugly syntax.

At the time, I assumed [PEP8](https://www.python.org/dev/peps/pep-0008/#package-and-module-names) would call for underscores, but now it seems better to break convention - a longer semantically meaningful name seems better than a short, all lower case name.   At which point, hyphens vs. underscores is a value judgement.

danizen commented:
So, I'll make the changes you suggest and let's see.

danizen committed:
Change name of package back to django-oracle-drcp.

- Consideration given to PEP8
- PEP8 says short, no underscores.
- Semantically rich name seems more important than convention.

danizen committed:
Changes to make pool configuration options clearer.

danizen committed:
Change README back to refer to `django-oracle-drcp`

danizen commented:
I'll test this on Monday, and maybe add a STATISTICS.md based on an email from work that compares CONN_MAX_AGE against Oracle DRCP.   The goal was to show that you are not giving up much performance by having a pool rather than dedicated per-process connections.

danizen committed:
Change variable name consistently to pool_config

Commit:
Revise Markdown documentation

- Change README to make 'POOL' optional
- Add PERFORMANCE readme containing some numbers

Commit:
Make the questions end with a question mark

danizen commented:
I tested with modifications from last night, and it works.  I think this is ready to merge, but still welcome suggested changes.
